ha/setup/cluster_layout.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+Module provides an interface for component files which describe HA cluster.
+"""
+
+import json
+from typing import Dict, List
+from abc import ABC
+from ha.core.error import HAError, HA_CLUSTER_CONFIG_ERROR
+
+
+class ClusterConfigError(HAError):
+    """
+    Error during cluster configuration import or validation.
+    """
+    def __init__(self, desc=None):
+        _desc = "HA cluster configuration error" if desc is None else desc
```

Comment:
I'm afraid you're assigning to a local variable that will be disposed as soon as `__init__` method is left.

Reply:
It is passed to `super().__init__ invocation`.